eng/tools/typespec-validation/src/rules/compile.ts

```diff
@@ -44,6 +45,14 @@ export class CompileRule implements Rule {
       errorOutput += stderr;
     }
 
+    const gitDiffResult = await gitDiffTopSpecFolder(host, folder);
```

Comment:
Don't call "git diff" if "tsp compile" or "tsp format" fails.  The result will probably be misleading or useless.